src/mprisbluezproxy.cpp

```diff
@@ -0,0 +1,592 @@
+        connect(m_currentController.data(), &MprisController::rateChanged, this, &MprisBluezProxy::rateChanged);
+        connect(m_currentController.data(), &MprisController::shuffleChanged, this, &MprisBluezProxy::shuffleChanged);
+        connect(m_currentController.data(), &MprisController::volumeChanged, this, &MprisBluezProxy::volumeChanged);
+        connect(m_currentController.data(), &MprisController::seeked, this, &MprisBluezProxy::seeked);
+
+        // Sync with the proxy player
+        syncData();
+    }
+}
+
+void MprisBluezProxy::disconnectController()
+{
+    qmlInfo(this) << "MprisBluezProxy disconnect controller";
+    if (!m_currentController.isNull()) {
+        // Mpris Root Interface
+        disconnect(m_currentController.data(), &MprisController::identityChanged, this, &MprisBluezProxy::identityChanged);
+
+        // Mpris Player Interface
+        disconnect(m_currentController.data(), &MprisController::canControlChanged, this, &MprisBluezProxy::canControlChanged);
+        disconnect(m_currentController.data(), &MprisController::canGoNextChanged, this, &MprisBluezProxy::canGoNextChanged);
+        disconnect(m_currentController.data(), &MprisController::canGoPreviousChanged, this, &MprisBluezProxy::canGoPreviousChanged);
+        disconnect(m_currentController.data(), &MprisController::canPauseChanged, this, &MprisBluezProxy::canPauseChanged);
+        disconnect(m_currentController.data(), &MprisController::canPlayChanged, this, &MprisBluezProxy::canPlayChanged);
+        disconnect(m_currentController.data(), &MprisController::canSeekChanged, this, &MprisBluezProxy::canSeekChanged);
+        disconnect(m_currentController.data(), &MprisController::loopStatusChanged, this, &MprisBluezProxy::loopStatusChanged);
+        disconnect(m_currentController.data(), &MprisController::maximumRateChanged, this, &MprisBluezProxy::maximumRateChanged);
+        disconnect(m_currentController.data(), &MprisController::metadataChanged, this, &MprisBluezProxy::metadataChanged);
+        disconnect(m_currentController.data(), &MprisController::minimumRateChanged, this, &MprisBluezProxy::minimumRateChanged);
+        disconnect(m_currentController.data(), &MprisController::playbackStatusChanged, this, &MprisBluezProxy::playbackStatusChanged);
+        disconnect(m_currentController.data(), &MprisController::positionChanged, this, &MprisBluezProxy::positionChanged);
+        disconnect(m_currentController.data(), &MprisController::rateChanged, this, &MprisBluezProxy::rateChanged);
+        disconnect(m_currentController.data(), &MprisController::shuffleChanged, this, &MprisBluezProxy::shuffleChanged);
+        disconnect(m_currentController.data(), &MprisController::volumeChanged, this, &MprisBluezProxy::volumeChanged);
+        disconnect(m_currentController.data(), &MprisController::seeked, this, &MprisBluezProxy::seeked);
```

Comment:
Unless we care about external connections, could just do disconnect(controller, 0, this, 0)?